File: Frontend/public/src/services/CourseService.js
import axios from "axios";

const COURSE_API_BASE_URL = 'http://localhost:8080/api/courses';

class CourseService {
    getAllCourses() {
        return axios.get(COURSE_API_BASE_URL);
    }

    getCourseById(id) {
        return axios.get(COURSE_API_BASE_URL + '/' + id);
    }

    createCourse(course) {
        return axios.post(COURSE_API_BASE_URL, course);
    }

    updateCourse(id, course) {
        return axios.put(COURSE_API_BASE_URL + '/' + id, course);
    }

    deleteCourse(id) {
        return axios.delete(COURSE_API_BASE_URL + '/' + id);
    }
}

export default new CourseService(); 

File: School-Management-System-2025/src/main/java/com/rehan/sms/services/CourseService.java
package com.rehan.sms.services;

import com.rehan.sms.dto.CourseDto;
import com.rehan.sms.exception.ResourceNotFoundException;
import java.util.List;

public interface CourseService {
    List<CourseDto> getAllCourses();

    CourseDto getCourseById(Long id) throws ResourceNotFoundException;

    CourseDto createCourse(CourseDto courseDto);

    CourseDto updateCourse(Long id, CourseDto courseDto) throws ResourceNotFoundException;

    void deleteCourse(Long id) throws ResourceNotFoundException;
}